component/src/test/java/io/siddhi/extension/io/nats/sink/NatsSinkAuthTestCase.java

```diff
@@ -0,0 +1,221 @@
+                + "destination='nats-test2', "
+                + "server.urls='" + "nats://localhost:" + port + "', "
+                + "auth.type='token', token='test')"
+                + "define stream inputStream (name string, age int, country string);";
+
+        natsClient.subscribe();
+        SiddhiAppRuntime executionPlanRuntime = siddhiManager.
+                createSiddhiAppRuntime(inStreamDefinition);
+        InputHandler inputStream = executionPlanRuntime.getInputHandler("inputStream");
+        executionPlanRuntime.start();
+
+        for (int i = 0; i < 20; i++) {
+            inputStream.send(new Object[]{"MIKE", i, "Germany"});
+        }
+        Thread.sleep(1000);
+        Assert.assertTrue(resultContainer.assertMessageContent("<events><event><name>MIKE</name><age>19</age>" +
+                "<country>Germany</country></event></events>"));
+        siddhiManager.shutdown();
+        simpleWebServer.close();
+    }
+
+    @Test(dependsOnMethods = "natsCorePublishWithToken")
+    public void natsStreamingWithToken() throws InterruptedException, TimeoutException, IOException {
+        startDocker(false, "nats-streaming", "--auth", "test");
+        ResultContainer resultContainer = new ResultContainer(2, 3);
+        STANClient stanClient = new STANClient("test-cluster", "stan_test1", "nats://localhost:"
+                + port, resultContainer);
+        stanClient.setToken("test".toCharArray());
+        stanClient.connect();
+        SiddhiManager siddhiManager = new SiddhiManager();
+
+        String inStreamDefinition = "@App:name('Test-plan1')\n"
+                + "@sink(type='nats', @map(type='xml'), "
+                + "destination='nats-test1', "
+                + "client.id='test-plan1-siddhi',"
+                + "server.urls='" + "nats://localhost:" + port + "', "
+                + "streaming.cluster.id='test-cluster', auth.type='token', token='test'"
+                + ")"
+                + "define stream inputStream (name string, age int, country string);";
+
+        stanClient.subsripeFromNow("nats-test1");
+        SiddhiAppRuntime executionPlanRuntime = siddhiManager.
+                createSiddhiAppRuntime(inStreamDefinition);
+        InputHandler inputStream = executionPlanRuntime.getInputHandler("inputStream");
+        executionPlanRuntime.start();
+
+        inputStream.send(new Object[]{"JAMES", 23, "USA"});
+        inputStream.send(new Object[]{"MIKE", 23, "Germany"});
+        Thread.sleep(500);
+        Assert.assertTrue(resultContainer.assertMessageContent("JAMES"));
+        Assert.assertTrue(resultContainer.assertMessageContent("MIKE"));
+        siddhiManager.shutdown();
+        simpleWebServer.close();
+    }
+
+    @Test(dependsOnMethods = "natsStreamingWithToken")
+    public void natsCorePublishWithTLS() throws Exception {
+        startDocker(true, "nats", "--tls", "--tlscert", "certs/server.pem", "--tlskey", "certs/key.pem");
```

Comment:
Do we need to start a docker container for each test case? Can’t we use a one Docker instance?

Reply:
In order to run each testcases nats server has to start with different authentication, We can not add multiple authentications into one server, that why I start docker container for each test case